src/tests/rules/no-undef.coffee

```diff
@@ -322,6 +322,33 @@ ruleTester.run 'no-undef', rule,
       message: "'Empty' is not defined."
       type: 'Identifier'
     ]
+  ,
+    # whitelist filename case insensitively (case-insensitive rule off)
+    code: '''
+      Empty2()
+    '''
+    output: '''
+      Empty2()
+    '''
+    errors: [
+      message: "'Empty2' is not defined."
+      type: 'Identifier'
+    ]
+  ,
+    # whitelist filename case insensitively (case-insensitive rule on)
+    code: '''
+      Empty2()
+    '''
+    output: '''
+      import Empty2 from 'fixtures/empty2'
+
+      Empty2()
+    '''
+    errors: [
+      message: "'Empty2' is not defined."
+      type: 'Identifier'
+    ]
+    settings: 'known-imports/case-insensitive': true
```

Comment:
I'd prefer a settings name that's more specific about *what* is case-insensitive - maybe `case-insensitive-whitelist-filenames`?

Or as your test comments suggest, `whitelist-filename-case-insensitive`?